src/sessionid.ts

```diff
@@ -25,31 +27,44 @@ export class SessionIdManager {
         this._sessionActivityTimestamp = null
 
         const persistenceName = config['persistence_name'] || config['token']
+        const desiredTimeout = config['session_idle_timeout_seconds'] || MAX_SESSION_IDLE_TIMEOUT
+
+        if (desiredTimeout > MAX_SESSION_IDLE_TIMEOUT) {
+            console.warn(
+                '[PostHog] session_idle_timeout_seconds cannot be  greater than 30 minutes. Using 30 minutes instead.'
+            )
+        } else if (desiredTimeout < MIN_SESSION_IDLE_TIMEOUT) {
+            console.warn(
+                '[PostHog] session_idle_timeout_seconds cannot be less than 60 seconds. Using 60 seconds instead.'
+            )
+        }
 
-        this.window_id_storage_key = 'ph_' + persistenceName + '_window_id'
-        this.primary_window_exists_storage_key = 'ph_' + persistenceName + '_primary_window_exists'
+        this._sessionTimeoutMs =
+            Math.min(Math.max(desiredTimeout, MIN_SESSION_IDLE_TIMEOUT), MAX_SESSION_IDLE_TIMEOUT) * 1000
```

Comment:
the `Math.min(Math.max` one-liner feels easy to misunderstand and break in future

I'd be tempted to put the whole thing in a getter and split the concepts 

But then we're adding lines of code and more LOC = more bugs so 🤷

Reply:
Middle ground is - I added tests ;)